test/cache.c

```diff
@@ -26856,20 +26577,16 @@ check_flush_deps_err(unsigned paged)
         } /* end switch */
 
         takedown_cache(file_ptr, false, false);
-        if (!pass)
+        if (!pass) {
+            file_ptr = NULL;
```

Comment:
Prevent accidentally calling `takedown_cache()` twice in case the first time fails